File: Examples/CSharp/Working-with-Text/GetPlainTextOfVisio.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

namespace Aspose.Diagram.Examples.CSharp.Working_with_Text
{
    public class GetPlainTextOfVisio
    {
        // ExStart:GetPlainTextOfVisio
        static string text = "";
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ShapeText();
            // Load diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");

            // Get Visio diagram page
            Aspose.Diagram.Page page = diagram.Pages.GetPage("Page-1");

            // Iterate through the shapes
            foreach (Aspose.Diagram.Shape shape in page.Shapes)
            {
                // Extract plain text from the shape
                GetShapeText(shape);
            }
            // Display extracted text
            Console.WriteLine(text);
        }
        private static void GetShapeText(Aspose.Diagram.Shape shape)
        {
            // Filter shape text
            if (shape.Text.Value.Text != "")
                text += Regex.Replace(shape.Text.Value.Text, "\\<.*?>", "");

            // For image shapes
            if (shape.Type == TypeValue.Foreign)
                text += (shape.Name);

            // For group shapes
            if (shape.Type == TypeValue.Group)
                foreach (Aspose.Diagram.Shape subshape in shape.Shapes)
                {
                    GetShapeText(subshape);
                }
        }
        // ExEnd:GetPlainTextOfVisio
    }
}

File: Examples/VisualBasic/Working-with-Text/GetPlainTextOfVisio.vb
Imports Aspose.Diagram
Imports System
Imports System.Text.RegularExpressions
Public Class GetPlainTextOfVisio
    ' ExStart:GetPlainTextOfVisio
    Shared text As String = ""
    Public Shared Sub Run()
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_ShapeText()
        ' Load diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))

        ' Get Visio diagram page
        Dim page As Aspose.Diagram.Page = diagram.Pages.GetPage("Page-1")

        ' Iterate through the shapes
        For Each shape As Aspose.Diagram.Shape In page.Shapes
            ' Extract plain text from the shape
            GetShapeText(shape)
        Next
        ' Display extracted text
        Console.WriteLine(text)
    End Sub
    Private Shared Sub GetShapeText(shape As Aspose.Diagram.Shape)
        ' Filter shape text
        If shape.Text.Value.Text <> "" Then
            text += Regex.Replace(shape.Text.Value.Text, "\<.*?>", "")
        End If

        ' For image shapes
        If shape.Type = TypeValue.Foreign Then
            text += (shape.Name)
        End If

        ' For group shapes
        If shape.Type = TypeValue.Group Then
            For Each subshape As Aspose.Diagram.Shape In shape.Shapes
                GetShapeText(subshape)
            Next
        End If
    End Sub
    ' ExEnd:GetPlainTextOfVisio
End Class
